### PR TITLE
A two way binding flows upstream through a CP.

### PR DESCRIPTION
**Pull request by rwjblue**
This is a rewritten version of #11225 to work with the new testing structure. 

Closes #11225.

### CODE REVIEW

**Comment by rwjblue** [comment]
I'd like @mixonic to confirm that this properly replaces https://github.com/emberjs/ember.js/pull/11225.

File: packages/ember-glimmer/tests/integration/components/curly-components-test.js [hunk 1]
```diff
@@ -956,4 +957,49 @@ moduleFor('Components test: curly components', class extends RenderingTest {
...
+          this._bar = value;
+          return this._bar;
+        }
+      })
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+
+      template: '{{bar}}'
```
Comment (mixonic) [inline packages/ember-glimmer/tests/integration/components/curly-components-test.js]: That the CP is consumed by the template is different than in https://github.com/emberjs/ember.js/pull/11225

Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/curly-components-test.js]: Gotcha, I will add another test without that consumption.

Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/curly-components-test.js]: Added another test for that (see below).

**Comment by rwjblue** [comment]
Added a second test that must be skipped (along with a small comment explaining when the regression was introduced). This test should be made to pass prior to enabling `ember-glimmer` (it should have been fixed in HTMLBars in 1.13, but that :ship: has sailed).

**Pull request merged**